pytest_plugins/logging_hooks.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+import logzero
+import pytest
+from xdist import get_xdist_worker_id
+from xdist import is_xdist_worker
+
+from robottelo.logging import DEFAULT_DATE_FORMAT
+from robottelo.logging import logger
+from robottelo.logging import robottelo_log_dir
+from robottelo.logging import robottelo_log_file
+
+try:
+    from pytest_reportportal import RPLogger
+    from pytest_reportportal import RPLogHandler
+except ImportError:
+    pass
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_sessionstart(session):
+    """Handle xdist and ReportPortal logging configuration at session start
+
+    Set up a separate logger for each pytest-xdist worker
+    if worker_id != 'master' then xdist is running in multi-threading so
+    a logfile named 'robottelo_gw{worker_id}.log' will be created.
+
+    Add a handler for ReportPortal logging
+    """
+    worker_id = get_xdist_worker_id(session)
+    worker_formatter = logzero.LogFormatter(
+        fmt=f'%(asctime)s - {worker_id} - %(name)s - %(levelname)s - %(message)s',
+        datefmt=DEFAULT_DATE_FORMAT,
+    )
+    use_rp_logger = getattr(session.config, '_reportportal_configured', False)
+    if use_rp_logger:
+        logging.setLoggerClass(RPLogger)
+
+    if is_xdist_worker(session):
+        if f'{worker_id}' not in [h.get_name() for h in logger.handlers]:
+            handlers_to_remove = [
+                h
+                for h in logger.handlers
+                if isinstance(h, logging.FileHandler)
+                and getattr(h, 'baseFilename', None) == str(robottelo_log_file)
+            ]
+            for handler in handlers_to_remove:
+                logger.removeHandler(handler)
+            file_handler = logging.FileHandler(
+                robottelo_log_dir.joinpath(f'robottelo_{worker_id}.log')
+            )
+            file_handler.set_name(f'robottelo.{worker_id}')
+            file_handler.setFormatter(worker_formatter)
+            logger.addHandler(file_handler)
+
+            if use_rp_logger:
```

Comment:
Still testing this piece, having a stability issue with our RP instance.

Reply:
I've got RP live logging working correctly now. I wasn't able to do it in a `pytest_sessionstart` hook, I had to do it in a fixture. But the logger doesn't have to be used from the fixture - the root logger imported through `robottelo.logging` will log to RP just fine.